pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -50,7 +50,7 @@ const (
 	// a machineconfig pool is going to be requeued:
 	//
 	// 5ms, 10ms, 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1.3s, 2.6s, 5.1s, 10.2s, 20.4s, 41s, 82s
-	maxRetries = 15
+	maxRetries = 18
```

Comment:
The kubelet config is dependent on the controller config here: https://github.com/harche/machine-config-operator/blob/f28b2c2a4102228af4b6754522e13384711969e7/pkg/controller/kubelet-config/kubelet_config_controller.go#L321-L341

Ideally, initial rollout would be nice, but that isn't how the feature works today. Changes to kubelet configurations require a reboot.

Reply:
Ah my bad. I missed that connection. I think this code does make sense to make the wait explicit. that said, I'm not sure if this will achieve the desired effect of waiting for the initial configuration of the pool to settle (the linked bugzilla), since

https://github.com/openshift/machine-config-operator/blob/5527f842eda1fd51538590a592d3c95a3cd679d9/pkg/controller/kubelet-config/kubelet_config_controller.go#L478

was blocking it before now anyways. The race between the KCC generating MCs from kubeletconfigs and the in-cluster MCC generating the initial node configuration still exists I think.

If we want to go this route, maybe https://github.com/openshift/machine-config-operator/blob/5527f842eda1fd51538590a592d3c95a3cd679d9/pkg/controller/kubelet-config/kubelet_config_controller.go#L466 we should instead check if the pool has a rendered config set to the pool's spec. Then the kubeletconfig should always get generated after an initial rendered configs gets generated (we want that, since it won't contain kubeletconfigs like bootstrap), and thus bypass the "not found" issue.

Alternatively, I  still think it might be a cleaner solution just to add a bootstrap reader mode to the kubeletconfigcontoller and  have the bootstrap MCC call it (achieving day 1), which if we want to do in the future, we'd have to revert our proposed fix here,

Reply:
Ideally, we need to update the code to run at bootstrap. Perhaps @QiWang19 can look into updating the KubeletConfig as a second pass. I would like to unblock Omer.

The node_controller does something similar to see if the pool is ready...
https://github.com/harche/machine-config-operator/blob/5527f842eda1fd51538590a592d3c95a3cd679d9/pkg/controller/node/node_controller.go#L720-L727